.github/workflows/atlantis-image.yml

```diff
@@ -158,3 +158,48 @@ jobs:
     runs-on: ubuntu-22.04
     steps:
       - run: 'echo "No build required"'
+
+  test:
+    needs: [changes]
+    if: needs.changes.outputs.should-run-build == 'true'
+    name: Test Image With Goss
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        image_type: [alpine, debian]
+    env:
+      # Set docker repo to either the fork or the main repo where the branch exists
+      DOCKER_REPO: ghcr.io/${{ github.repository }}
+
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@d70bba72b1f3fd22344832f00baa16ece964efeb # v3
+        # https://github.com/docker/build-push-action/issues/761#issuecomment-1575006515
+        with:
+          driver-opts: |
+            image=moby/buildkit:v0.13.1
+
+      - name: "Build and load into Docker"
+        if: contains(fromJson('["push", "pull_request"]'), github.event_name)
+        uses: docker/build-push-action@2cdde995de11925a030ce8070c3d77a52ffcf1c0 # v5
+        with:
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          context: .
+          build-args: |
+            ATLANTIS_BASE_TAG_TYPE=${{ matrix.image_type }}
+          push: false
+          load: true
+          tags: "${{ env.DOCKER_REPO }}:goss-test"
+          target: ${{ matrix.image_type }}
+
+      - name: "Setup Goss"
+        uses: e1himself/goss-installation-action@fbb6fb55d3e59c96045b2500eeb8ce0995d99ac1 # v1.2.1
+        with:
+          version: "v0.4.6"
+
+      - name: Execute Goss tests
```

Comment:
How come this runs in a separate job instead of in the same build job? My concern is that the build of the container takes a long time and this additional build will double the time for builds (even tho it runs in parallel)

Reply:
It runs here for a couple of reasons.

1. This step doesn't do multi-arch builds (so it's _much_ faster, in this PR, Debian took 26s, and Alpine took ~60s)
1. The current `build` job doesn't support exporting the image to Docker - at least, I couldn't figure out how to make it work; BuildKit kept complaining about not exporting multi-arch to Docker or a tarball.
  1. Since `dgoss` requires the image to be available in Docker (or a registry) to pull it, we would either need to export it to Docker (which I couldn't get working) or push it for every build so it can be pulled (didn't love this)

Reply:
For the individual checks they ran in 2min this time around :)